Tests/test_generic.py

```diff
@@ -0,0 +1,127 @@
+from io import BytesIO
+
+import pytest
+
+from PyPDF2.constants import TypFitArguments as TF
+from PyPDF2.errors import PdfReadError, PdfStreamError
+from PyPDF2.generic import (
+    ArrayObject,
+    BooleanObject,
+    Destination,
+    FloatObject,
+    IndirectObject,
+    NameObject,
+    NullObject,
+    NumberObject,
+    PdfObject,
+    createStringObject,
+    readHexStringFromStream,
+    readStringFromStream,
+)
+
+
+def test_float_object_exception():
+    assert FloatObject("abc") == 0
+
+
+def test_number_object_exception():
+    with pytest.raises(OverflowError):
+        NumberObject(1.5 * 2**10000)
+
+
+def test_createStringObject_exception():
+    with pytest.raises(TypeError) as exc:
+        createStringObject(123)
+    assert exc.value.args[0] == "createStringObject should have str or unicode arg"
+
+
+@pytest.mark.parametrize("value, expected", [(b"true", b"true"), (b"false", b"false")])
+def test_boolean_object(value, expected):
+    stream = BytesIO(value)
+    BooleanObject.readFromStream(stream)
```

Comment:
We should have an assert on what this returns

Reply:
> We should have an assert on what this returns?

Good point, I completely missed that it returns something :+1:

Reply:
> capture that it leaves the incoming stream at the end of it

Also a good idea. I've just added it :+1: